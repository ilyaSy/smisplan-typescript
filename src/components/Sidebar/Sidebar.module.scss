.sidebar {
  width: 60px;
  height: 100%;
}

.menu-item {
  margin-top: 10px !important;
  padding-top: 10px !important;
  padding-bottom: 40px !important;
}

.menu-submenu {
  margin-top: 10px !important;
  padding-top: 10px !important;
}

.add-icon {
  transition: all 1s ease-in-out !important;

  &:hover {
    transform: rotate(360deg);
  }
}

.add-icon:hover {
  
}