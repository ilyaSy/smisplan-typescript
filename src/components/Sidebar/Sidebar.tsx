import { memo } from "react";

const Sidebar: React.FC = () => {
  return (
    <></>
  )
}

export default memo(Sidebar);