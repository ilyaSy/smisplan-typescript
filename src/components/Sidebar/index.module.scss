.sidebar {
  width: 60px;
  height: 100%;
  padding-right: 0px !important;

  &__menu-item {
    margin-top: 10px !important;
    padding-right: 0px !important;
    padding-top: 10px !important;
    padding-bottom: 40px !important;
  }

  &__menu-submenu {
    margin-top: 10px !important;
    padding-top: 10px !important;
  }

  &__menu-submenu > div {
    text-overflow: unset !important;
    padding-right: 0px !important;
  }

  &__add-icon {
    transition: all 1s ease-in-out !important;

    &:hover {
      transform: rotate(360deg);
    }
  }
}
