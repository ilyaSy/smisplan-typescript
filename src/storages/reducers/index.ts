import { combineReducers } from "redux"
import metadataReducer from "./metadata";
import dataReducer from "./data";

const reducers = combineReducers({
  dataReducer,
  metadataReducer
})

export default reducers;