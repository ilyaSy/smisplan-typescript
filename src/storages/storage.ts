import { applyMiddleware, combineReducers, createStore } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import thunk from "redux-thunk";
import { dataReducer } from "./reducers/data";
import { metadataReducer } from "./reducers/metadata";

export const storage = createStore(combineReducers({
  dataReducer,
  metadataReducer
}), composeWithDevTools(applyMiddleware(thunk)));
