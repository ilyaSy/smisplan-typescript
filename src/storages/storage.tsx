import { applyMiddleware, createStore } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import { apiMiddleware } from "./middleware/apiMiddleware";
import reducers from "./reducers";

export const storage = createStore(reducers, composeWithDevTools(applyMiddleware(thunk, apiMiddleware)));

export const StorageProvider: React.FC = ({ children }) => {
  return (
    <Provider store={storage}>
      {children}
    </Provider>
  );
}
