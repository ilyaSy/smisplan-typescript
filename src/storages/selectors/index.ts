import { createSelector } from '@reduxjs/toolkit';
import { useSelector } from 'react-redux';

import { TRootState } from 'storages/storage';

export const dataSelector = () => createSelector((state: TRootState) => state.dataReducer, (data) => data);

export const useMetadataSelector = () => useSelector((state: TRootState) => state.metadataReducer);
