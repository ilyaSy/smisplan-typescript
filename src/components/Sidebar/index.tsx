import React, { memo, useEffect, useState } from 'react';
import { useDispatch } from "react-redux";
import { Menu, Tooltip } from 'antd';
import {
  SettingOutlined,
  PlusCircleTwoTone,
} from '@ant-design/icons';
import DataAddModal from '../Modals/DataAddModal';
import {dataAddAction} from "../../storages/actions/data";
import { useGetTablename } from "../../utils/hooks/useGetTablename";
import { useDataSelector } from "../../storages/selectors/data";
import { PrintPDF } from '../UI/PrintPDF';
import classes from './Sidebar.module.scss';

const Sidebar: React.FC = () => {
  const dispatch = useDispatch();
  const [sidebarAction, setSidebarAction] = useState<string>('');

  const tablename = useGetTablename();

  const {isError: isErrorData, isLoading: isLoadingData} = useDataSelector();

  const handleAddData = () => setSidebarAction('addData');

  const handleCloseModal = () => setSidebarAction('');

  useEffect(() => {
    if (!isErrorData && !isLoadingData) handleCloseModal();
  }, [isErrorData, isLoadingData])

  return (
    <Menu
      mode='vertical'
      theme='dark'
      className={classes.sidebar}
      selectedKeys={[sidebarAction]}
    >
      <DataAddModal
        isOpen={sidebarAction === 'addData'}
        onAddHandler={(data) => {
          dispatch(dataAddAction(tablename, data));
        }}
        onClose={handleCloseModal}
      />

      <Menu.Item key='addData' className={classes['sidebar__menu-item']}>
        <Tooltip placement='bottomLeft' title='Добавить' className='tooltip'>
          <PlusCircleTwoTone
            className={classes['sidebar__add-icon']}
            style={{ fontSize: '30px' }}
            twoToneColor='#52c41a'
            onClick={handleAddData}
          />
        </Tooltip>
      </Menu.Item>

      <Menu.Item key='printPDF' className={classes['sidebar__menu-item']}>
        <PrintPDF />
      </Menu.Item>

      <Menu.SubMenu key='settings'
        className={classes['sidebar__menu-submenu']}
        icon={<SettingOutlined style={{ fontSize: '30px' }} />}
      >
        <Menu.Item key='color-scheme'>Цветовая тема</Menu.Item>
        <Menu.Item key='manual'>Инструкция</Menu.Item>
      </Menu.SubMenu>
    </Menu>
  );
};

export default memo(Sidebar);
