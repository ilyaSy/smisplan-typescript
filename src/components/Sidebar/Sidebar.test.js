import { configure, render, screen, fireEvent, waitFor } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { StorageProvider } from '../../storages/storage';
import Sidebar from '.';

configure({
  testIdAttribute: 'id',
});

describe('Sidebar', () => {
  test('Displaying correctly', async () => {
    render(
      <BrowserRouter>
        <StorageProvider>
          <Sidebar />
        </StorageProvider>
      </BrowserRouter>
    );

    await waitFor(() => {
      const sidebarElements = screen.getAllByRole('menuitem');
      expect(sidebarElements.length).toBe(3);
      expect(sidebarElements[0]).toHaveAttribute('data-menu-id', 'rc-menu-uuid-test-addData');
      expect(sidebarElements[1]).toHaveAttribute('data-menu-id', 'rc-menu-uuid-test-printPDF');
      expect(sidebarElements[2]).toHaveAttribute('data-menu-id', 'rc-menu-uuid-test-settings');
    });
  });

  test.skip('Show add modal menu', async () => {
    render(
      <BrowserRouter>
        <StorageProvider>
          <Sidebar />
        </StorageProvider>
      </BrowserRouter>
    );

    fireEvent.click(screen.getAllByRole('menuitem')[0]);
    await waitFor(() => {
      expect(screen.getByRole('document')).toHaveClass('ant-modal-wrap');
    });
  });

  test('Show settings submenu', async () => {
    render(
      <BrowserRouter>
        <StorageProvider>
          <Sidebar />
        </StorageProvider>
      </BrowserRouter>
    );

    fireEvent.mouseOver(screen.getAllByRole('menuitem')[2]);
    await waitFor(() => {
      expect(screen.getByText('Цветовая тема')).toBeInTheDocument();
      expect(screen.getByText('Инструкция')).toBeInTheDocument();
    });
  });
});
