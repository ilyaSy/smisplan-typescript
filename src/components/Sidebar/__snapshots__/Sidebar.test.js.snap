// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="ant-menu ant-menu-root ant-menu-vertical ant-menu-dark sidebar"
        data-menu-list="true"
        role="menu"
        tabindex="0"
      >
        <li
          class="ant-menu-item ant-menu-item-only-child sidebar__menu-item"
          data-menu-id="rc-menu-uuid-test-addData"
          role="menuitem"
          tabindex="-1"
        >
          <span
            class="ant-menu-title-content"
          >
            <span
              aria-label="plus-circle"
              class="anticon anticon-plus-circle sidebar__add-icon tooltip"
              role="img"
              style="font-size: 30px;"
              tabindex="-1"
            >
              <svg
                aria-hidden="true"
                data-icon="plus-circle"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                  fill="#52c41a"
                />
                <path
                  d="M512 140c-205.4 0-372 166.6-372 372s166.6 372 372 372 372-166.6 372-372-166.6-372-372-372zm192 396c0 4.4-3.6 8-8 8H544v152c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V544H328c-4.4 0-8-3.6-8-8v-48c0-4.4 3.6-8 8-8h152V328c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v152h152c4.4 0 8 3.6 8 8v48z"
                  fill="#f6ffed"
                />
                <path
                  d="M696 480H544V328c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v152H328c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h152v152c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V544h152c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8z"
                  fill="#52c41a"
                />
              </svg>
            </span>
          </span>
        </li>
        <li
          class="ant-menu-item ant-menu-item-only-child sidebar__menu-item"
          data-menu-id="rc-menu-uuid-test-printPDF"
          role="menuitem"
          tabindex="-1"
        >
          <span
            class="ant-menu-title-content"
          >
            <span
              aria-label="printer"
              class="anticon anticon-printer tooltip"
              role="img"
              style="font-size: 30px;"
            >
              <svg
                aria-hidden="true"
                data-icon="printer"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M820 436h-40c-4.4 0-8 3.6-8 8v40c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-40c0-4.4-3.6-8-8-8zm32-104H732V120c0-4.4-3.6-8-8-8H300c-4.4 0-8 3.6-8 8v212H172c-44.2 0-80 35.8-80 80v328c0 17.7 14.3 32 32 32h168v132c0 4.4 3.6 8 8 8h424c4.4 0 8-3.6 8-8V772h168c17.7 0 32-14.3 32-32V412c0-44.2-35.8-80-80-80zM360 180h304v152H360V180zm304 664H360V568h304v276zm200-140H732V500H292v204H160V412c0-6.6 5.4-12 12-12h680c6.6 0 12 5.4 12 12v292z"
                />
              </svg>
            </span>
          </span>
        </li>
        <li
          class="ant-menu-submenu ant-menu-submenu-vertical sidebar__menu-submenu"
          role="none"
        >
          <div
            aria-controls="rc-menu-uuid-test-settings-popup"
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-menu-submenu-title"
            data-menu-id="rc-menu-uuid-test-settings"
            role="menuitem"
            tabindex="-1"
          >
            <span
              aria-label="setting"
              class="anticon anticon-setting ant-menu-item-icon"
              role="img"
              style="font-size: 30px;"
            >
              <svg
                aria-hidden="true"
                data-icon="setting"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M924.8 625.7l-65.5-56c3.1-19 4.7-38.4 4.7-57.8s-1.6-38.8-4.7-57.8l65.5-56a32.03 32.03 0 009.3-35.2l-.9-2.6a443.74 443.74 0 00-79.7-137.9l-1.8-2.1a32.12 32.12 0 00-35.1-9.5l-81.3 28.9c-30-24.6-63.5-44-99.7-57.6l-15.7-85a32.05 32.05 0 00-25.8-25.7l-2.7-.5c-52.1-9.4-106.9-9.4-159 0l-2.7.5a32.05 32.05 0 00-25.8 25.7l-15.8 85.4a351.86 351.86 0 00-99 57.4l-81.9-29.1a32 32 0 00-35.1 9.5l-1.8 2.1a446.02 446.02 0 00-79.7 137.9l-.9 2.6c-4.5 12.5-.8 26.5 9.3 35.2l66.3 56.6c-3.1 18.8-4.6 38-4.6 57.1 0 19.2 1.5 38.4 4.6 57.1L99 625.5a32.03 32.03 0 00-9.3 35.2l.9 2.6c18.1 50.4 44.9 96.9 79.7 137.9l1.8 2.1a32.12 32.12 0 0035.1 9.5l81.9-29.1c29.8 24.5 63.1 43.9 99 57.4l15.8 85.4a32.05 32.05 0 0025.8 25.7l2.7.5a449.4 449.4 0 00159 0l2.7-.5a32.05 32.05 0 0025.8-25.7l15.7-85a350 350 0 0099.7-57.6l81.3 28.9a32 32 0 0035.1-9.5l1.8-2.1c34.8-41.1 61.6-87.5 79.7-137.9l.9-2.6c4.5-12.3.8-26.3-9.3-35zM788.3 465.9c2.5 15.1 3.8 30.6 3.8 46.1s-1.3 31-3.8 46.1l-6.6 40.1 74.7 63.9a370.03 370.03 0 01-42.6 73.6L721 702.8l-31.4 25.8c-23.9 19.6-50.5 35-79.3 45.8l-38.1 14.3-17.9 97a377.5 377.5 0 01-85 0l-17.9-97.2-37.8-14.5c-28.5-10.8-55-26.2-78.7-45.7l-31.4-25.9-93.4 33.2c-17-22.9-31.2-47.6-42.6-73.6l75.5-64.5-6.5-40c-2.4-14.9-3.7-30.3-3.7-45.5 0-15.3 1.2-30.6 3.7-45.5l6.5-40-75.5-64.5c11.3-26.1 25.6-50.7 42.6-73.6l93.4 33.2 31.4-25.9c23.7-19.5 50.2-34.9 78.7-45.7l37.9-14.3 17.9-97.2c28.1-3.2 56.8-3.2 85 0l17.9 97 38.1 14.3c28.7 10.8 55.4 26.2 79.3 45.8l31.4 25.8 92.8-32.9c17 22.9 31.2 47.6 42.6 73.6L781.8 426l6.5 39.9zM512 326c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm79.2 255.2A111.6 111.6 0 01512 614c-29.9 0-58-11.7-79.2-32.8A111.6 111.6 0 01400 502c0-29.9 11.7-58 32.8-79.2C454 401.6 482.1 390 512 390c29.9 0 58 11.6 79.2 32.8A111.6 111.6 0 01624 502c0 29.9-11.7 58-32.8 79.2z"
                />
              </svg>
            </span>
            <span
              class="ant-menu-title-content"
            />
            <i
              class="ant-menu-submenu-arrow"
            />
          </div>
        </li>
        <li
          class="ant-menu-item ant-menu-item-only-child sidebar__menu-item"
          data-menu-id="rc-menu-uuid-test-resetFilters"
          role="menuitem"
          tabindex="-1"
        >
          <span
            class="ant-menu-title-content"
          >
            <span
              aria-label="eye-invisible"
              class="anticon anticon-eye-invisible tooltip"
              role="img"
              style="font-size: 30px;"
            >
              <svg
                aria-hidden="true"
                data-icon="eye-invisible"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                />
                <path
                  d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                />
              </svg>
            </span>
          </span>
        </li>
      </ul>
      <div
        aria-hidden="true"
        style="display: none;"
      />
    </div>
  </body>,
  "container": <div>
    <ul
      class="ant-menu ant-menu-root ant-menu-vertical ant-menu-dark sidebar"
      data-menu-list="true"
      role="menu"
      tabindex="0"
    >
      <li
        class="ant-menu-item ant-menu-item-only-child sidebar__menu-item"
        data-menu-id="rc-menu-uuid-test-addData"
        role="menuitem"
        tabindex="-1"
      >
        <span
          class="ant-menu-title-content"
        >
          <span
            aria-label="plus-circle"
            class="anticon anticon-plus-circle sidebar__add-icon tooltip"
            role="img"
            style="font-size: 30px;"
            tabindex="-1"
          >
            <svg
              aria-hidden="true"
              data-icon="plus-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                fill="#52c41a"
              />
              <path
                d="M512 140c-205.4 0-372 166.6-372 372s166.6 372 372 372 372-166.6 372-372-166.6-372-372-372zm192 396c0 4.4-3.6 8-8 8H544v152c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V544H328c-4.4 0-8-3.6-8-8v-48c0-4.4 3.6-8 8-8h152V328c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v152h152c4.4 0 8 3.6 8 8v48z"
                fill="#f6ffed"
              />
              <path
                d="M696 480H544V328c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v152H328c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h152v152c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V544h152c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8z"
                fill="#52c41a"
              />
            </svg>
          </span>
        </span>
      </li>
      <li
        class="ant-menu-item ant-menu-item-only-child sidebar__menu-item"
        data-menu-id="rc-menu-uuid-test-printPDF"
        role="menuitem"
        tabindex="-1"
      >
        <span
          class="ant-menu-title-content"
        >
          <span
            aria-label="printer"
            class="anticon anticon-printer tooltip"
            role="img"
            style="font-size: 30px;"
          >
            <svg
              aria-hidden="true"
              data-icon="printer"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M820 436h-40c-4.4 0-8 3.6-8 8v40c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-40c0-4.4-3.6-8-8-8zm32-104H732V120c0-4.4-3.6-8-8-8H300c-4.4 0-8 3.6-8 8v212H172c-44.2 0-80 35.8-80 80v328c0 17.7 14.3 32 32 32h168v132c0 4.4 3.6 8 8 8h424c4.4 0 8-3.6 8-8V772h168c17.7 0 32-14.3 32-32V412c0-44.2-35.8-80-80-80zM360 180h304v152H360V180zm304 664H360V568h304v276zm200-140H732V500H292v204H160V412c0-6.6 5.4-12 12-12h680c6.6 0 12 5.4 12 12v292z"
              />
            </svg>
          </span>
        </span>
      </li>
      <li
        class="ant-menu-submenu ant-menu-submenu-vertical sidebar__menu-submenu"
        role="none"
      >
        <div
          aria-controls="rc-menu-uuid-test-settings-popup"
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          data-menu-id="rc-menu-uuid-test-settings"
          role="menuitem"
          tabindex="-1"
        >
          <span
            aria-label="setting"
            class="anticon anticon-setting ant-menu-item-icon"
            role="img"
            style="font-size: 30px;"
          >
            <svg
              aria-hidden="true"
              data-icon="setting"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M924.8 625.7l-65.5-56c3.1-19 4.7-38.4 4.7-57.8s-1.6-38.8-4.7-57.8l65.5-56a32.03 32.03 0 009.3-35.2l-.9-2.6a443.74 443.74 0 00-79.7-137.9l-1.8-2.1a32.12 32.12 0 00-35.1-9.5l-81.3 28.9c-30-24.6-63.5-44-99.7-57.6l-15.7-85a32.05 32.05 0 00-25.8-25.7l-2.7-.5c-52.1-9.4-106.9-9.4-159 0l-2.7.5a32.05 32.05 0 00-25.8 25.7l-15.8 85.4a351.86 351.86 0 00-99 57.4l-81.9-29.1a32 32 0 00-35.1 9.5l-1.8 2.1a446.02 446.02 0 00-79.7 137.9l-.9 2.6c-4.5 12.5-.8 26.5 9.3 35.2l66.3 56.6c-3.1 18.8-4.6 38-4.6 57.1 0 19.2 1.5 38.4 4.6 57.1L99 625.5a32.03 32.03 0 00-9.3 35.2l.9 2.6c18.1 50.4 44.9 96.9 79.7 137.9l1.8 2.1a32.12 32.12 0 0035.1 9.5l81.9-29.1c29.8 24.5 63.1 43.9 99 57.4l15.8 85.4a32.05 32.05 0 0025.8 25.7l2.7.5a449.4 449.4 0 00159 0l2.7-.5a32.05 32.05 0 0025.8-25.7l15.7-85a350 350 0 0099.7-57.6l81.3 28.9a32 32 0 0035.1-9.5l1.8-2.1c34.8-41.1 61.6-87.5 79.7-137.9l.9-2.6c4.5-12.3.8-26.3-9.3-35zM788.3 465.9c2.5 15.1 3.8 30.6 3.8 46.1s-1.3 31-3.8 46.1l-6.6 40.1 74.7 63.9a370.03 370.03 0 01-42.6 73.6L721 702.8l-31.4 25.8c-23.9 19.6-50.5 35-79.3 45.8l-38.1 14.3-17.9 97a377.5 377.5 0 01-85 0l-17.9-97.2-37.8-14.5c-28.5-10.8-55-26.2-78.7-45.7l-31.4-25.9-93.4 33.2c-17-22.9-31.2-47.6-42.6-73.6l75.5-64.5-6.5-40c-2.4-14.9-3.7-30.3-3.7-45.5 0-15.3 1.2-30.6 3.7-45.5l6.5-40-75.5-64.5c11.3-26.1 25.6-50.7 42.6-73.6l93.4 33.2 31.4-25.9c23.7-19.5 50.2-34.9 78.7-45.7l37.9-14.3 17.9-97.2c28.1-3.2 56.8-3.2 85 0l17.9 97 38.1 14.3c28.7 10.8 55.4 26.2 79.3 45.8l31.4 25.8 92.8-32.9c17 22.9 31.2 47.6 42.6 73.6L781.8 426l6.5 39.9zM512 326c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm79.2 255.2A111.6 111.6 0 01512 614c-29.9 0-58-11.7-79.2-32.8A111.6 111.6 0 01400 502c0-29.9 11.7-58 32.8-79.2C454 401.6 482.1 390 512 390c29.9 0 58 11.6 79.2 32.8A111.6 111.6 0 01624 502c0 29.9-11.7 58-32.8 79.2z"
              />
            </svg>
          </span>
          <span
            class="ant-menu-title-content"
          />
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
      <li
        class="ant-menu-item ant-menu-item-only-child sidebar__menu-item"
        data-menu-id="rc-menu-uuid-test-resetFilters"
        role="menuitem"
        tabindex="-1"
      >
        <span
          class="ant-menu-title-content"
        >
          <span
            aria-label="eye-invisible"
            class="anticon anticon-eye-invisible tooltip"
            role="img"
            style="font-size: 30px;"
          >
            <svg
              aria-hidden="true"
              data-icon="eye-invisible"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
              />
              <path
                d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
              />
            </svg>
          </span>
        </span>
      </li>
    </ul>
    <div
      aria-hidden="true"
      style="display: none;"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
